DOC: Clarify development workflow (#9126)
astropy/astropy: docs/development/workflow/development_workflow.rst

those changes in your local copy, commit them to your local repo and push them
to GitHub. GitHub will automatically update your pull request.

.. _no git pull:

> Do not create a merge commit

Reviewer: ```suggestion
Do Not Create a Merge Commit
```

****************************

If your branch associated with the pull request falls behind the ``master``

> branch of https://github.com/astropy/astropy , GitHub might offer you option

Reviewer: ```suggestion
branch of https://github.com/astropy/astropy, GitHub might offer you the option
```

> to catch up or resolve conflict via its web interface; do not use them. Using

Reviewer: ```suggestion
to catch up or resolve conflicts via its web interface, but do not use this. Using
```

> them might create a "merge commit" in your commit history, which is

Reviewer: ```suggestion
the web interface might create a "merge commit" in your commit history, which is
```

undesirable, as a "merge commit" can introduce maintenance overhead for the

> release manager. Do not use ``git pull`` command either.

Reviewer 2: "and mess up the branch structure". Not sure how to say that in a document friendly way.

Author: Me neither. A typical contributor might not care about preserving branch structure anyway, but I think we need to convey the *impact* of such a thing happening. Suggestions welcome!

Reviewer 2: Maybe: `a "merge commit" introduces undesirable branch structure complexity.` @eteq might want to chime in.
Bottom line is that one merge commit is not the end of the word, but usually it escalates and having two usually means that the PR became a spagetti mess, and not rebasable any  more only with much larger effort. Keeping things clean and tidy is not a bad policy.

Author: @bsipocz , your suggestion was added as part of copyediting by @lglattly . FYI. Thank you both!

> release manager. Do not use ``git pull`` command either.

Reviewer: ```suggestion
release manager as well as undesirable branch structure complexity. Do not use the ``git pull`` command either.
```

> You are more likely to run into *conflicts* here--places where the changes you

Reviewer: ```suggestion
You are more likely to run into *conflicts* here — places where the changes you
```

> made conflict with changes that someone else made--than anywhere else. Ask for

Reviewer: ```suggestion
made conflict with changes that someone else made — than anywhere else. Ask for
```
